src/coreComponents/python/modules/geosx_mesh_tools_package/geosx_mesh_tools/abaqus_converter.py

```diff
@@ -0,0 +1,136 @@
+
+import meshio
+from meshio._mesh import CellBlock
+import numpy as np
+import argparse
+import logging
+
+
+def convert_abaqus_to_gmsh(input_mesh, output_mesh, logger=None):
+    """
+    @brief Convert an abaqus mesh to gmsh 2 format, preserving nodeset information.
+    @details If the code encounters any issues with region/element indices,
+             the conversion will attempt to continue, with errors
+             indicated by -1 values in the output file.
+    @param input_mesh path of the input abaqus file
+    @param output_mesh path of the output gmsh file
+    @param logger an instance of logging.Logger
+    """
+    # Initialize the logger if it is empty
+    if not logger:
+      logging.basicConfig(level=logging.WARNING)
+      logger = logging.getLogger(__name__)
+
+    # Load the mesh
+    logger.info('Reading abaqus mesh...')
+    mesh = meshio.Mesh.read(input_mesh, "abaqus")
+
+    # Convert the element regions to tags
+    logger.info('Converting region tags...')
+    region_list = list(mesh.cell_sets.keys())
+    n_regions = len(region_list)
+    cell_ids = np.zeros(len(mesh.cells[0][1]), dtype=int) - 1
+    for ii, region in enumerate(region_list):
+        mesh.field_data[region] = [ii + 1, 3]
+        cell_ids[mesh.cell_sets[region][0]] = ii + 1
+
+    # Add to the meshio datastructure
+    mesh.cell_data['gmsh:physical'] = [cell_ids]
+    mesh.cell_data['gmsh:geometrical'] = [cell_ids]
+    if (-1 in cell_ids):
+        logger.warning('Some element regions did not convert correctly to tags!')
+        logger.warning('Note: These will be indicated by a -1 in the output file.')
+
+    # Build the face elements
+    logger.info('Converting nodesets to face elements, tags...')
+    new_tris, tri_nodeset, tri_region = [], [], []
+    new_quads, quad_nodeset, quad_region = [], [], []
+
+    for nodeset_id, nodeset_name in enumerate(mesh.point_sets):
+        logger.info('  %s' % (nodeset_name))
+        mesh.field_data[nodeset_name] = [nodeset_id + n_regions + 1, 2]
+        nodeset = mesh.point_sets[nodeset_name]
+
+        # Search the elements
+        for element_id in range(len(mesh.cells[0][1])):
```

Comment:
Sorry to chime in, I am not particularly familiar with Abaqus format, but here, you are assuming that your mesh will always be represented by a single block. Is it always true?

Also, it's more pythonic to use ``enumerate``:

```python
for element_id, cellblock in enumerate(mesh.cells[0].data):
```

and replace all subsequent ``mesh.cells[0][1][element_id]`` by ``cellblock``.

Reply:
No problem at all!  Yes, abaqus supports multi-block formats, but the specific meshes that I had developed this for were all monolithic.  I definitely plan to incorporate this before merging.